plugins/modules/tanzu_mission_control_secret.py

```diff
@@ -0,0 +1,798 @@
+
+
+def handle_logic(secret_status, wished_state):
+    if secret_status and wished_state == "present":
+        return "do_nothing"
+
+    if secret_status and wished_state == "update":
+        return "force_update"
+
+    if secret_status and wished_state == "absent":
+        return "delete"
+
+    if not secret_status and wished_state == "present" or wished_state == "update":
+        return "create"
+
+    if not secret_status and wished_state == "absent":
+        return "do_nothing"
+
+    return "error"
+
+
+def load_module():
+    argument_spec = dict(
+        api_host=dict(type="str", required=True),
+        api_token=dict(type="str", required=True, no_log=True),
+        management_cluster_name=dict(type="str"),
+        provisioner_name=dict(type="str"),
+        cluster_name=dict(type="str"),
+        cluster_namespace=dict(type="str", required=True),
+        cluster_group=dict(type="str"),
+        secret_name=dict(type="str", required=True),
+        registry_host=dict(type="str"),
+        registry_username=dict(type="str"),
+        registry_password=dict(type="str", no_log=True),
+        secret_type=dict(
+            type="str",
+            choices=["opaque", "docker_config"],
+            default="opaque",
+        ),
+        data=dict(type="dict", default={}),
+        export=dict(type="bool", default=False),
+        state=dict(
+            type="str", default="present", choices=["present", "absent", "update"]
+        ),
+    )
+
+    return AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ["cluster_group", "cluster_name"],
+            ["cluster_group", "management_cluster_name"],
+            ["cluster_group", "provisioner_name"],
+            ["data", "registry_host"],
+            ["data", "registry_username"],
+            ["data", "registry_password"],
+        ],
+        required_one_of=[
+            ["cluster_name", "cluster_group"],
+            ["data", "registry_host"],
+            ["data", "registry_username"],
+            ["data", "registry_password"],
+        ],
+        required_together=[
+            ["cluster_name", "management_cluster_name", "provisioner_name"],
+            ["registry_host", "registry_username", "registry_password"],
+        ],
+    )
+
+
+def create_dockerconfig(registry_host, registry_username, registry_password):
+    payload = {
+        "auths": {
+            registry_host: {
+                "username": registry_username,
+                "password": registry_password,
+                "auth": b64encode(
+                    "{}:{}".format(registry_username, registry_password).encode()
+                ).decode(),
+            }
+        }
+    }
+    return {".dockerconfigjson": json.dumps(payload)}
+
+
+def main():
+    module = load_module()
+
+    def key_or_none(param_name):
+        if param_name in module.params:
```

Comment:
This condition is never `False`. The dictionary `module.params` **always** contains **all** the module parameters, even when they are not passed to the module and have no default value, in which case the value for that key is `None`.

From what I see of the use of this function below, you can simply remove this function and access the parameters directly.

Reply:
I think I was doing something wrong then with my local version of it, since I remember the `params` field not being populated with all the parameters.. I might have missed a declaration in the `AnsibleModule` or something, hence the reason why I wrote that little helper :-)